### PR TITLE
test(atomic): Add more cypress tests for sort & pager components

### PR DESCRIPTION
**Pull request by lvu285**
https://coveord.atlassian.net/browse/KIT-507
Sort
- Add check for URL
- Add check for sort with field QRE generated correctly
- Add more check for invalid sort-criteria field

Pager
- Add check for 'disable-navigation' button when go to page 2

Facet
- Remove async when function is not async/await

### CODE REVIEW

File: packages/atomic/cypress/integration/sort-dropdown.cypress.ts [hunk 1]
```diff
@@ -70,14 +67,42 @@ describe('Sort Dropdown Component', () => {
...
+describe('When sort with invalid criteria option', () => {
+  it('Should render an error when <atomic-sort-criteria> is invalid', () => {
     setUpPage(`
-      <atomic-sort-dropdown>
-        <atomic-sort-criteria</atomic-sort-criteria>
-      </atomic-sort-dropdown>
-    `);
-    cy.wait(500);
+        <atomic-sort-dropdown>
+          <atomic-sort-criteria</atomic-sort-criteria>
```
Comment (olamothe) [inline packages/atomic/cypress/integration/sort-dropdown.cypress.ts:81]: What's the difference between this test and the one on line 89-91 ? It seems very similar ? 🤔 

Both has no `criteria` defined.

File: packages/atomic/cypress/integration/facet-utils.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export async function assertSortCriteria(
     '@coveoSearch',
     requestBodyOrder
   );
-  const firstRequestBodyFacets = requestBody.facets[0];
+  const firstRequestBodyFacets = (requestBody.facets as any)[0];
```
Comment (olamothe) [inline packages/atomic/cypress/integration/facet-utils.ts:55]: As we talked in the channel this morning, I don't think we should continue to do tests that verify the request itself, but should focus on the end results we get (aka: facets are in the correct order).

In any case, why did you need to modify this here, with a cast to `any` ?

**Review by olamothe — APPROVED** [review]

File: packages/atomic/cypress/integration/pager.cypress.ts [hunk 3]
```diff
@@ -122,6 +122,15 @@ describe('Pager Test Suites', () => {
       cy.get('@pagerLi').get(PagerSelectors.buttonPrevious).should('not.exist');
     });
 
+    it('should load without navigation buttons when user goes to page 2', () => {
```
Comment (samisayegh) [inline packages/atomic/cypress/integration/pager.cypress.ts:125]: Either the test description is unclear or I am confused about the test.

Why does the navigation disappear on page 2? Shouldn't it be absent on all pages when `enable-navigation-buttons=false` is configured? What is special about page 2 in this case?

File: packages/atomic/cypress/integration/sort-dropdown.cypress.ts [hunk 4]
```diff
@@ -70,14 +67,42 @@ describe('Sort Dropdown Component', () => {
...
+        <atomic-sort-dropdown>
+          <atomic-sort-criteria criteria="size ascending; date ascending"></atomic-sort-criteria>
+        </atomic-sort-dropdown>
+      `);
+    shouldRenderErrorComponent(sortDropdown);
+  });
+});
+
+describe('When sort has no label', () => {
+  it('Should display default label');
```
Comment (samisayegh) [inline packages/atomic/cypress/integration/sort-dropdown.cypress.ts:107]: Should this test be implemented in this PR or a future one?

**Review by samisayegh — APPROVED** [review]

File: packages/atomic/cypress/integration/sort-dropdown.cypress.ts (hunk 1 above)
Comment (lvu285) [reply packages/atomic/cypress/integration/sort-dropdown.cypress.ts:81]: ha, really, I found it different. But nevermind, I will modify, to have the test without `<atomic-sort-dropdown>` then

File: packages/atomic/cypress/integration/facet-utils.ts (hunk 2 above)
Comment (lvu285) [reply packages/atomic/cypress/integration/facet-utils.ts:55]: We have 1 jira to review all the existing check `requestBody` later
I put as any, just to remove the red warning check for the code ,

File: packages/atomic/cypress/integration/pager.cypress.ts (hunk 3 above)
Comment (lvu285) [reply packages/atomic/cypress/integration/pager.cypress.ts:125]: Normally when you load the page, the 'previous' button won't be visible, so we go to page 2 is to make sure that even if we activate/use the 'pager', the "previous button" won't be visible at all

File: packages/atomic/cypress/integration/sort-dropdown.cypress.ts (hunk 4 above)
Comment (lvu285) [reply packages/atomic/cypress/integration/sort-dropdown.cypress.ts:113]: no, in the future one, the real fix is not done yet
